src/core/control/Control.cpp

```diff
@@ -2130,40 +2130,82 @@ auto Control::openFile(fs::path filepath, int scrollToPage, bool forceOpen) -> b
     if ((loadedDocument != nullptr && loadHandler.isAttachedPdfMissing()) ||
         !loadHandler.getMissingPdfFilename().empty()) {
         // give the user a second chance to select a new PDF filepath, or to discard the PDF
-
         const fs::path missingFilePath = fs::path(loadHandler.getMissingPdfFilename());
-        const std::string msg1 =
-                FS(_F("The attached background file {1} could not be found. It might have been moved, renamed or "
-                      "deleted.\nIt was last seen at: {2}") %
-                   missingFilePath.filename().string() % missingFilePath.parent_path().string());
-        const std::string msg2 =
-                FS(_F("The background file {1} could not be found. It might have been moved, renamed or deleted.\nIt "
-                      "was last seen at: {2}") %
-                   missingFilePath.filename().string() % missingFilePath.parent_path().string());
-        GtkWidget* dialog =
-                gtk_message_dialog_new(getGtkWindow(), GTK_DIALOG_MODAL, GTK_MESSAGE_QUESTION, GTK_BUTTONS_NONE, "%s",
-                                       loadHandler.isAttachedPdfMissing() ? msg1.c_str() : msg2.c_str());
 
-        gtk_dialog_add_button(GTK_DIALOG(dialog), _("Select another PDF"), 1);
-        gtk_dialog_add_button(GTK_DIALOG(dialog), _("Remove PDF Background"), 2);
-        gtk_dialog_add_button(GTK_DIALOG(dialog), _("Cancel"), 3);
+        string parentFolderPath;
+        string filename;
+
+        // since POSIX systems detect the whole Windows path as a filename, this checks whether missingFilePath contains
+        // a Windows path
+        GMatchInfo* matchInfo = nullptr;
+        GRegex* regex = g_regex_new(R"([A-Z]:\\.*\\(.*))", static_cast<GRegexCompileFlags>(0),
+                                    static_cast<GRegexMatchFlags>(0), nullptr);
+        g_regex_match(regex, (gchar*) missingFilePath.filename().c_str(), static_cast<GRegexMatchFlags>(0), &matchInfo);
+
+        if (g_match_info_matches(matchInfo)) {
+            parentFolderPath = missingFilePath.filename().string();
+            filename = g_match_info_fetch(matchInfo, 1);
+        } else {
+            parentFolderPath = missingFilePath.parent_path().string();
+            filename = missingFilePath.filename().string();
+        }
+
+        std::string msg;
+        if (loadHandler.isAttachedPdfMissing()) {
+            msg = FS(_F("The attached background file \"{1}\" could not be found. It might have been moved, "
+                        "renamed or deleted.") %
+                     filename);
+        } else {
+            msg = FS(_F("The background file \"{1}\" could not be found. It might have been moved, renamed or "
+                        "deleted.") %
+                     filename);
+        }
+
+        msg += FS(_F("\nIt was last seen at: \"{1}\"") % parentFolderPath);
```

Comment:
I would say that these two messages are very similar. One of the two should be enough, since the average user won't know the difference between a "attached" file and not attached file.
```suggestion
        auto msg = FS(_F("The background file \"{1}\" could not be found. It might have been moved, "
                         "renamed or deleted.\nIt was last seen at: \"{2}\"") %
                      filename % parentFolderPath);
```

Reply:
I agree the average user might not know, but it would help whenever the user submits a bug report. I'd keep it like that.